apps/src/code-studio/components/progress/TeacherStageInfo.jsx

```diff
@@ -79,13 +98,28 @@ const TeacherStageInfo = React.createClass({
             </span>
           }
           {lockable && <StageLock stage={stage}/>}
+          {experiments.isEnabled('hiddenStages') && <div style={styles.toggle}>
+            <HiddenStageToggle
+              hidden={this.props.isHidden}
+              onChange={this.onClickHiddenToggle}
+            />
+          </div>
+          }
         </div>
       </div>
     );
   }
 });
 
-export default connect(state => ({
-  hasNoSections: state.stageLock.sectionsLoaded &&
-    Object.keys(state.stageLock.sections).length === 0
+export default connect((state, ownProps) => {
+  const isHidden = state.hiddenStage[ownProps.stage.id];
+  return {
+    isHidden,
+    hasNoSections: state.stageLock.sectionsLoaded &&
+      Object.keys(state.stageLock.sections).length === 0
+    };
+}, dispatch => ({
```

Comment:
I think I prefer the latter.  We haven't really talked about how much preprocessing we want in our connect methods.  At the point where you're using `ownProps` it seems a little like there's some 'business logic' and I worry it'll be a little hidden down in the connect methods.